src/ng/animate.js

```diff
@@ -237,16 +238,16 @@ var $AnimateProvider = ['$provide', function($provide) {
    */
   this.classNameFilter = function(expression) {
     if (arguments.length === 1) {
-      this.$$classNameFilter = (expression instanceof RegExp) ? expression : null;
-      if (this.$$classNameFilter) {
-        var reservedRegex = new RegExp("(\\s+|\\/)" + NG_ANIMATE_CLASSNAME + "(\\s+|\\/)");
-        if (reservedRegex.test(this.$$classNameFilter.toString())) {
-          throw $animateMinErr('nongcls','$animateProvider.classNameFilter(regex) prohibits accepting a regex value which matches/contains the "{0}" CSS class.', NG_ANIMATE_CLASSNAME);
-
+      classNameFilter = (expression instanceof RegExp) ? expression : null;
```

Comment:
This is not about ignoring invalid expressions (in the sense that they are of the wrong type).
This check is throwing when using `ng-animate` in your (valid) RegExp, since `ng-animate` is added by Angular itself.
Not sure why this is important tbh. (Off the top of my head, `.ng-animate` will be added **after** the `classNameFilter()` but I might be wrong.)